lib/rgeo/geos/capi_factory.rb

```diff
@@ -182,10 +182,10 @@ def marshal_dump  # :nodoc:
           'hasm' => (_flags & 0x4 != 0),
           'srid' => _srid,
           'bufr' => _buffer_resolution,
-          'wktg' => _wkt_generator._properties,
-          'wkbg' => _wkb_generator._properties,
-          'wktp' => _wkt_parser._properties,
-          'wkbp' => _wkb_parser._properties,
+          'wktg' => _wkt_generator ? _wkt_generator._properties : {},
+          'wkbg' => _wkb_generator ? _wkb_generator._properties : {},
+          'wktp' => _wkt_parser ? _wkt_parser._properties : {},
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.